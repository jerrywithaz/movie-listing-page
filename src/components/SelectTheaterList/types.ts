import { Cinema, SelectedCinema } from "core/types";

export type SelectTheaterListProps = {
    onTheaterSelected: (theater: Cinema) => void,
    theaters: Cinema[],
    selectedTheater: SelectedCinema
} & React.HTMLAttributes<HTMLDivElement>;

export type TheaterProps = {
    selected: boolean
} & React.HTMLAttributes<HTMLButtonElement>;