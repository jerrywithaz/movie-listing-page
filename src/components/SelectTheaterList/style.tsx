import React from 'react';
import styled from 'styled-components';
import { Heading2 as DefaultHeading2 } from 'components/Heading';
import { TheaterProps } from './types';

export const SelectTheaterList = styled.div`
    display: flex;
    flex-direction: column;
    max-height: 196px;
    margin-bottom: ${props => props.theme.spacing.large}px;
    overflow: hidden;
`;

export const Heading2 = styled(DefaultHeading2)`
    ${props => `
        margin-bottom: ${props.theme.spacing.small}px;
    `}
`;

export const Theaters = styled.div`
    display: flex;
    flex-wrap: wrap;
    flex-flow: row wrap;
    justify-content: flex-start;
    flex: 1;
    overflow: auto;
`;

export const Theater = styled(({
    selected, ...rest
}: TheaterProps) => 
    <button {...rest}/>
)`
    ${props => `
        background: ${props.selected ? `${props.theme.palette.primary}`: `${props.theme.palette.common.white}`};
        border: 1px solid ${props.selected ? `${props.theme.palette.primary}`: `${props.theme.palette.common.black}`};
        border-radius: ${props.theme.borderRadius.default}px;
        color: ${props.selected ? `${props.theme.palette.common.white}`: `${props.theme.palette.common.black}`};
        height: 40px;
        padding: 0px ${props.theme.spacing.small}px;
        margin: ${props.theme.spacing.xsmall}px 0px;
        width: 176px;
        letter-spacing: ${props.theme.letterSpacing.small};
        text-transform: uppercase;
        text-align: center;
        font-size: 16px;
        font-weight: bold;
        white-space: nowrap;
        min-width: 0px;
        display: flex;
        align-items: center;
        justify-content: center;
        &:focus {
            outline: none;
        }
        &:hover, &:focus {
            cursor: pointer;
            background: ${props.theme.palette.primary};
            border: 1px solid ${props.theme.palette.primary};
            color: ${props.theme.palette.common.white};
        }
        &:nth-child(3n - 1) {
            margin-left: ${props.theme.spacing.small}px;
            margin-right: ${props.theme.spacing.small}px;
        }
        
    `}
    ${props => props.theme.media.sm`
        margin-right: ${props.theme.spacing.small}px;
        width: calc(33.33% - 16px);
        &:nth-child(3n - 1) {
            margin-left: 0px;
        }
    `}
    ${props => props.theme.media.xs`
        width: 100%;
        margin: 0px 0px ${props.theme.spacing.small}px 0px;
        &:nth-child(3n - 1) {
            margin: 0px 0px ${props.theme.spacing.small}px 0px;
        }
    `}
    
`;