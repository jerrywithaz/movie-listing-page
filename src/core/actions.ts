import { 
    ActionType, 
    RequestAustinFeedAction, 
    RequestAustinFeedSuccessAction, 
    RequestAustinFeedFailureAction 
} from './types';

export function requestAustinFeed(): RequestAustinFeedAction {
    return {
        type: ActionType.REQUEST_AUSTIN_FEED
    };
}

export function requestAustinFeedSuccess(
    payload: object
): RequestAustinFeedSuccessAction {
    return {
        type: ActionType.REQUEST_AUSTIN_FEED_SUCCESS,
        payload: payload
    };
}

export function requestAustinFeedFailure(
    error: Error
): RequestAustinFeedFailureAction{
    return {
        type: ActionType.REQUEST_AUSTIN_FEED_FAILURE,
        error: error
    };
}