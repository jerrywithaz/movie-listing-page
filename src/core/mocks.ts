import { Cinema, Film, Session } from './types';

/**
 * Mock Data for Stories, Tests and any other development purposes
 */

export function getMockFilms(): Film[] {
    return [{
        "slug": "birds-of-prey",
        "title": "BIRDS OF PREY (AND THE FANTABULOUS EMANCIPATION OF ONE HARLEY QUINN)",
    },
    {
        "slug": "event-1917",
        "title": "1917",
    },
    {
        "slug": "the-gentlemen",
        "title": "THE GENTLEMEN",
    }];
};

export function getMockFilm(): Film {
    return {
        "slug": "birds-of-prey",
        "title": "BIRDS OF PREY (AND THE FANTABULOUS EMANCIPATION OF ONE HARLEY QUINN)",
    };
}

export function getMockSessions(): Session[] {
    return [{
        "cinemaId": "0003",
        "sessionId": "35751",
        "filmSlug": "birds-of-prey",
        "filmName": "BIRDS OF PREY",
    }, {
        "cinemaId": "0002",
        "sessionId": "35865",
        "filmSlug": "cats-rowdy-screening",
        "filmName": "CATS Rowdy Screening",
    }, {
        "cinemaId": "0002",
        "sessionId": "35751",
        "filmSlug": "birds-of-prey",
        "filmName": "BIRDS OF PREY"
    }, {
        "cinemaId": "0004",
        "sessionId": "160776",
        "filmSlug": "the-gentlemen",
        "filmName": "THE GENTLEMEN",
    }];
}

export function getMockTheaters(): Cinema[] {
    return [{
        "id": "0003",
        "slug": "village",
        "name": "Village",
        "status": "OPEN"
    }, {
        "id": "0004",
        "slug": "south-lamar",
        "name": "South Lamar",
        "status": "OPEN",
    }, {
        "id": "0002",
        "slug": "ritz",
        "name": "Ritz",
        "status": "OPEN"
    }, {
        "id": "0008",
        "slug": "mueller",
        "name": "Mueller",
        "status": "OPEN"
    }];
}

export function getMockTheater(): Cinema {
    return {
        "id": "0003",
        "slug": "village",
        "name": "Village",
        "status": "OPEN"
    };
}