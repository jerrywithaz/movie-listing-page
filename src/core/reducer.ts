import produce, { Draft } from 'immer';