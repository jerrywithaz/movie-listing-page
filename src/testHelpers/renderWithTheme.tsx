import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from 'containers/ThemeProvider';

function renderWithTheme(component: JSX.Element) {
    return renderer.create(
        <ThemeProvider>
          {component}
        </ThemeProvider>
      );
}

export default renderWithTheme;