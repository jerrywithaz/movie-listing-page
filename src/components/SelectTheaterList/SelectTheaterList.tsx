import React from 'react';
import sortBy from 'lodash/sortBy';
import { Cinema } from 'core/types';
import { SelectTheaterListProps } from './types';

import * as Styled from './style';

const SelectTheaterList = ({
    onTheaterSelected,
    theaters,
    selectedTheater,
    ...rest
}: SelectTheaterListProps) => {

    const theatersSorted = sortBy(theaters, "name");

    return (
        <Styled.SelectTheaterList {...rest}>
            <Styled.Heading2>Select Theater</Styled.Heading2>
            <Styled.Theaters>
                {theatersSorted.map((theater: Cinema) => {
                    return (
                        <Styled.Theater 
                            key={theater.id}
                            id={theater.slug} 
                            title={theater.name}
                            onClick={() => onTheaterSelected(theater)}
                            selected={selectedTheater?.id === theater.id}>
                                {theater.name}
                        </Styled.Theater>
                    );
                })}
            </Styled.Theaters>
        </Styled.SelectTheaterList>
    );

};

export default SelectTheaterList;