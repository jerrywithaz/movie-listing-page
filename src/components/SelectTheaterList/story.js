import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import SelectTheaterList from './';
import { getMockTheaters } from 'core/mocks';

const story = storiesOf('SelectTheaterList', module);

const theaters = getMockTheaters();

story.add('Default', () => (
    <SelectTheaterList 
        onTheaterSelected={action("Theater Selected")}
        theaters={theaters}/>
));
  