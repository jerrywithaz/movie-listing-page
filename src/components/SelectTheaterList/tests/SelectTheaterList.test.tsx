import React from 'react';
import SelectTheaterList from './../';
import renderWithTheme from 'testHelpers/renderWithTheme';
import { getMockTheaters } from 'core/mocks';

describe("components/SelectTheaterList", () => {

    it("renders without theaters", () => {

        const tree = renderWithTheme(
            <SelectTheaterList
                onTheaterSelected={() => {}} 
                theaters={[]} 
                selectedTheater={null}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();

    });

    it("renders with theaters", () => {

        const theaters = getMockTheaters();
        const selectedTheater = theaters[0];

        const tree = renderWithTheme(
            <SelectTheaterList
                onTheaterSelected={() => {}} 
                theaters={theaters} 
                selectedTheater={selectedTheater}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();

    });

});