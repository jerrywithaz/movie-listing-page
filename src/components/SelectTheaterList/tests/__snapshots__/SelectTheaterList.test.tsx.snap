// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SelectTheaterList renders with theaters 1`] = `
.c1 {
  font-size: 16px;
  line-height: 20px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  color: #6B6B6B;
  text-transform: uppercase;
  margin: 24px 0px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 196px;
  margin-bottom: 32px;
  overflow: hidden;
}

.c2 {
  margin-bottom: 16px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

.c4 {
  background: #ffffff;
  border: 1px solid #000000;
  border-radius: 4px;
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 40px;
  padding: 0px 16px;
  margin: 8px 0px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-weight: bold;
  text-transform: uppercase;
  white-space: nowrap;
  width: 176px;
}

.c4:hover,
.c4:focus {
  cursor: pointer;
  background: #F0B323;
  border: 1px solid #F0B323;
  color: #ffffff;
}

.c4:focus {
  outline: none;
}

.c4:nth-child(3n - 1) {
  margin-left: 16px;
  margin-right: 16px;
}

.c5 {
  background: #F0B323;
  border: 1px solid #F0B323;
  border-radius: 4px;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 40px;
  padding: 0px 16px;
  margin: 8px 0px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-weight: bold;
  text-transform: uppercase;
  white-space: nowrap;
  width: 176px;
}

.c5:hover,
.c5:focus {
  cursor: pointer;
  background: #F0B323;
  border: 1px solid #F0B323;
  color: #ffffff;
}

.c5:focus {
  outline: none;
}

.c5:nth-child(3n - 1) {
  margin-left: 16px;
  margin-right: 16px;
}

@media (max-width:768px) {
  .c4 {
    width: calc(50% - 16px);
    margin: 0px 16px 16px 0px;
  }

  .c4:nth-child(3n - 1) {
    margin: 0px 16px 16px 0px;
  }
}

@media (max-width:480px) {
  .c4 {
    width: 100%;
    margin: 0px 0px 16px 0px;
  }

  .c4:nth-child(3n - 1) {
    margin: 0px 0px 16px 0px;
  }
}

@media (max-width:768px) {
  .c5 {
    width: calc(50% - 16px);
    margin: 0px 16px 16px 0px;
  }

  .c5:nth-child(3n - 1) {
    margin: 0px 16px 16px 0px;
  }
}

@media (max-width:480px) {
  .c5 {
    width: 100%;
    margin: 0px 0px 16px 0px;
  }

  .c5:nth-child(3n - 1) {
    margin: 0px 0px 16px 0px;
  }
}

<section
  className="c0"
>
  <h2
    className="c1 c2"
  >
    Select Theater
  </h2>
  <div
    className="c3"
  >
    <button
      className="c4"
      id="mueller"
      onClick={[Function]}
      onKeyUp={[Function]}
      title="Mueller"
    >
      Mueller
    </button>
    <button
      className="c4"
      id="ritz"
      onClick={[Function]}
      onKeyUp={[Function]}
      title="Ritz"
    >
      Ritz
    </button>
    <button
      className="c4"
      id="south-lamar"
      onClick={[Function]}
      onKeyUp={[Function]}
      title="South Lamar"
    >
      South Lamar
    </button>
    <button
      className="c5"
      id="village"
      onClick={[Function]}
      onKeyUp={[Function]}
      title="Village"
    >
      Village
    </button>
  </div>
</section>
`;

exports[`components/SelectTheaterList renders without theaters 1`] = `
.c1 {
  font-size: 16px;
  line-height: 20px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  color: #6B6B6B;
  text-transform: uppercase;
  margin: 24px 0px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 196px;
  margin-bottom: 32px;
  overflow: hidden;
}

.c2 {
  margin-bottom: 16px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

<section
  className="c0"
>
  <h2
    className="c1 c2"
  >
    Select Theater
  </h2>
  <div
    className="c3"
  />
</section>
`;
